### PR TITLE
Add correct host info for cloud provider virtual machines

### PR DESCRIPTION
**Pull request by kubasobon**
By default, beats are expected to fill the `host` section with data of a machine the beat/agent runs on. This PR changes the logic by disabling the default processor providing the data and enriching select resources with asset data fetched from cloud providers.

:warning: It is worth noting that adding the `add_host_metadata` processor cloudbeat config file (e.g. `data/cloudbeat.yml`) will result in the old behavior.

### Summary of your changes

Changes made:
- drop Host processor from the pipeline
- publish `host.*` data for Azure VMs
- publish `host.*` data for GCP CloudCompute Instances

Please note the change will affect findings for rules related to virtual machine resources:
- CIS Azure 7.1 - 7.4
- CIS GCP 4.1 - 4.9, 4.11

All other findings will have an empty `host` section.

### Screenshot/Data

<details>
<summary>Data before the change</summary>

image

</details>

<details>
<summary>Data after the change</summary>

image

</details>

### Related Issues

Related to https://github.com/elastic/cloudbeat/issues/2081

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

No

### CODE REVIEW

**Comment by kubasobon** [comment]
Might be of use: https://github.com/elastic/cloudbeat/pull/2138

File: cmd/root.go [hunk 1]
```diff
@@ -36,7 +37,16 @@ import (
...
 // RootCmd to handle beats cli
-var RootCmd = cmd.GenRootCmdWithSettings(beater.New, instance.Settings{Name: Name, Version: version.CloudbeatSemanticVersion()})
+var RootCmd = cmd.GenRootCmdWithSettings(
+	beater.New,
+	instance.Settings{
+		Name:    Name,
+		Version: version.CloudbeatSemanticVersion(),
+		// Supply our own processing pipeline. Same as processing.MakeDefaultBeatSupport, but without
+		// `processing.WithHost`.
+		Processing: processing.MakeDefaultSupport(true, nil, processing.WithECS, processing.WithAgentMeta()),
```
Comment (kubasobon) [inline cmd/root.go:47]: CNVM will still get `host` information thanks to https://github.com/elastic/cloudbeat/blob/8f3098173baa57e49c34b4cd43742896b0b79a95/internal/vulnerability/events_creator.go#L254-L265

File: _meta/config/processors.yml.tmpl [hunk 2]
```diff
@@ -3,7 +3,8 @@
 # Configure processors to enhance or manipulate events generated by the beat.
 
 processors:
-  - add_host_metadata: ~
+# in case you run in EKS/Kubernetes environment, you can use the following processor to add node metadata
+# - add_host_metadata: ~
```
Comment (oren-zohar) [inline _meta/config/processors.yml.tmpl:7]: What would be the `host.*`/`user.*` values in KSPM/EKS where this processor is not enabled?

File: deploy/aws/cloudbeat-aws.yml [hunk 3]
```diff
@@ -47,7 +47,7 @@ output.elasticsearch:
 
 # ================================= Processors =================================
 processors:
-  - add_host_metadata: ~
+  # - add_host_metadata: ~
```
Comment (oren-zohar) [inline deploy/aws/cloudbeat-aws.yml:50]: for CSPM let's remove it completely, no? Does it make sense to re-enable it here?

File: internal/resources/fetching/fetchers/azure/assets_fetcher_test.go [hunk 4]
```diff
@@ -220,7 +220,11 @@ func (s *AzureAssetsFetcherTestSuite) TestFetcher_Fetch() {
 
 			ecs, err := result.GetElasticCommonData()
 			s.Require().NoError(err)
-			s.Empty(ecs)
+			if expected.Type == inventory.VirtualMachineAssetType {
```
Comment (oren-zohar) [inline internal/resources/fetching/fetchers/azure/assets_fetcher_test.go:223]: maybe we can add tests to other cases where the "properties.osProfile.computerName"  nested data is partly/completely missing

File: internal/resources/fetching/fetchers/gcp/assets_fetcher.go [hunk 5]
```diff
@@ -148,6 +148,22 @@ func (r *GcpAsset) GetMetadata() (fetching.ResourceMetadata, error) {
...
 func (r *GcpAsset) GetElasticCommonData() (map[string]any, error) {
+	if r.Type == fetching.CloudCompute && r.SubType == inventory.ComputeInstanceAssetType {
+		m := map[string]any{}
+		data := r.ExtendedAsset.Resource.Data
+		if data == nil {
+			return nil, nil
+		}
+		nameField, ok := data.Fields["name"]
+		if ok && nameField.GetStringValue() != "" {
+			m["host.name"] = nameField.GetStringValue()
```
Comment (oren-zohar) [inline internal/resources/fetching/fetchers/gcp/assets_fetcher.go:159]: nit: double `GetStringValue` but not a big deal
```suggestion
			nameValue := nameField.GetStringValue()
			if nameValue != "" {
				m["host.name"] = nameValue
			}
```

File: _meta/config/processors.yml.tmpl (hunk 2 above)
Comment (kubasobon) [reply _meta/config/processors.yml.tmpl:7]: I have added a function which always enables host metadata for KSPM. See https://github.com/elastic/cloudbeat/pull/2126/commits/3292c21a25a2f22b9f76a810b9882f8427917b07

**Review by moukoublen — APPROVED** [review]

**Comment by moukoublen** [comment]
LGTM 🚀

**Pull request merged**